vpc_endpoints.tf

```diff
@@ -2,6 +2,186 @@
+  vpc_endpoint_type = "Interface"
+  vpc_id            = aws_vpc.the_vpc.id
+}
+resource "aws_vpc_endpoint" "ssmmessages" {
+  provider = aws.sharedservicesprovisionaccount
+
+  private_dns_enabled = true
+  security_group_ids = [
+    aws_security_group.ssm_agent_endpoint.id,
+    aws_security_group.ssm_endpoint.id,
+  ]
+  service_name      = "com.amazonaws.${var.aws_region}.ssmmessages"
+  vpc_endpoint_type = "Interface"
+  vpc_id            = aws_vpc.the_vpc.id
+}
+
+# Interface endpoints associated with the CloudWatch agent
+resource "aws_vpc_endpoint" "logs" {
+  provider = aws.sharedservicesprovisionaccount
+
+  private_dns_enabled = true
+  security_group_ids = [
+    aws_security_group.cloudwatch_agent_endpoint.id,
+  ]
+  service_name      = "com.amazonaws.${var.aws_region}.logs"
+  vpc_endpoint_type = "Interface"
+  vpc_id            = aws_vpc.the_vpc.id
+}
+resource "aws_vpc_endpoint" "monitoring" {
+  provider = aws.sharedservicesprovisionaccount
+
+  private_dns_enabled = true
+  security_group_ids = [
+    aws_security_group.cloudwatch_agent_endpoint.id,
+  ]
+  service_name      = "com.amazonaws.${var.aws_region}.monitoring"
+  vpc_endpoint_type = "Interface"
+  vpc_id            = aws_vpc.the_vpc.id
+}
+
+# Associate the STS interface endpoint with the private subnets
+resource "aws_vpc_endpoint_subnet_association" "sts" {
+  provider = aws.sharedservicesprovisionaccount
+
+  for_each = toset(var.private_subnet_cidr_blocks)
+
+  subnet_id       = module.private.subnets[each.value].id
+  vpc_endpoint_id = aws_vpc_endpoint.sts.id
+}
+
+# Associate the SSM interface endpoints with the private subnets
+#
+# Note that SSM requires several other endpoints to function properly.
+# See here for more details:
+# https://docs.aws.amazon.com/systems-manager/latest/userguide/setup-create-vpc.html#sysman-setting-up-vpc-create
+resource "aws_vpc_endpoint_subnet_association" "ssm" {
+  provider = aws.sharedservicesprovisionaccount
+
+  for_each = toset(var.private_subnet_cidr_blocks)
+
+  subnet_id       = module.private.subnets[each.value].id
+  vpc_endpoint_id = aws_vpc_endpoint.ssm.id
+}
+resource "aws_vpc_endpoint_subnet_association" "ec2" {
+  provider = aws.sharedservicesprovisionaccount
+
+  for_each = toset(var.private_subnet_cidr_blocks)
+
+  subnet_id       = module.private.subnets[each.value].id
+  vpc_endpoint_id = aws_vpc_endpoint.ec2.id
+}
+resource "aws_vpc_endpoint_subnet_association" "ec2messages" {
+  provider = aws.sharedservicesprovisionaccount
+
+  for_each = toset(var.private_subnet_cidr_blocks)
+
+  subnet_id       = module.private.subnets[each.value].id
+  vpc_endpoint_id = aws_vpc_endpoint.ec2messages.id
+}
+resource "aws_vpc_endpoint_subnet_association" "kms" {
+  provider = aws.sharedservicesprovisionaccount
+
+  for_each = toset(var.private_subnet_cidr_blocks)
+
+  subnet_id       = module.private.subnets[each.value].id
+  vpc_endpoint_id = aws_vpc_endpoint.kms.id
+}
```

Comment:
Unless there's a specific reason for this ordering I think it should mirror the order of the `aws_vpc_endpoint` definitions.

```suggestion
resource "aws_vpc_endpoint_subnet_association" "ec2" {
  provider = aws.sharedservicesprovisionaccount

  for_each = toset(var.private_subnet_cidr_blocks)

  subnet_id       = module.private.subnets[each.value].id
  vpc_endpoint_id = aws_vpc_endpoint.ec2.id
}
resource "aws_vpc_endpoint_subnet_association" "ec2messages" {
  provider = aws.sharedservicesprovisionaccount

  for_each = toset(var.private_subnet_cidr_blocks)

  subnet_id       = module.private.subnets[each.value].id
  vpc_endpoint_id = aws_vpc_endpoint.ec2messages.id
}
resource "aws_vpc_endpoint_subnet_association" "kms" {
  provider = aws.sharedservicesprovisionaccount

  for_each = toset(var.private_subnet_cidr_blocks)

  subnet_id       = module.private.subnets[each.value].id
  vpc_endpoint_id = aws_vpc_endpoint.kms.id
}
resource "aws_vpc_endpoint_subnet_association" "ssm" {
  provider = aws.sharedservicesprovisionaccount

  for_each = toset(var.private_subnet_cidr_blocks)

  subnet_id       = module.private.subnets[each.value].id
  vpc_endpoint_id = aws_vpc_endpoint.ssm.id
}
```

Reply:
@jsf9k I see you resolved this but the change was not committed and there's no explanatory comment for why the change should not be implemented.

Reply:
Apologies, I don't know what happened there.